### PR TITLE
Add deployment helper

### PR DESCRIPTION
**Pull request by mcardosos**
PTAL
@jhendrixMSFT @marstr

### CODE REVIEW

File: arm/helpers/helpers.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package helpers
```
Comment (jhendrixMSFT) [inline arm/helpers/helpers.go]: I think we should give this package a more meaningful name, something that describes its use.  I'm bad at naming but will throw a few out here: template, deployment.

File: arm/helpers/helpers.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+)
+
+// BuildDeployment is a helper method that create a deployment object
```
Comment (jhendrixMSFT) [inline arm/helpers/helpers.go]: I think this needs to be beefed up (why would somebody want to call this method) and include links about ARM templates.  Also describe the return params, e.g. what do I actually do with the returned deployment object?

**Comment by jhendrixMSFT** [comment]
A more general comment, this creates some data structures but doesn't actually do anything in Azure.  Do we need another method that simply takes a template file and deploys it in Azure?

File: arm/helpers/helpers.go (hunk 1 above)
Comment (mcardosos) [reply arm/helpers/helpers.go]: Fixed

File: arm/helpers/helpers.go (hunk 2 above)
Comment (mcardosos) [reply arm/helpers/helpers.go]: Fixed

File: arm/deployment/deployment.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"io/ioutil"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+)
+
+// BuildDeployment is a helper that creates a resources.Deployment, which can
+// be used as a parameter for a CreateOrUpdate deployment operation.
+// templateFile is a local Azure template.
+// See https://github.com/Azure-Samples/resource-manager-go-template-deployment
+func BuildDeployment(mode resources.DeploymentMode, templateFile string, parameters map[string]interface{}) (deployment resources.Deployment, err error) {
```
Comment (jhendrixMSFT) [inline arm/deployment/deployment.go]: Since the package name is deployment I think we can change this function name to Build.

**Pull request merged**